src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -433,6 +433,22 @@ int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringR
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
         osStatus = SecTrustEvaluate(trust, &trustResult);
+
+        if (trustResult == kSecTrustResultRecoverableTrustFailure && osStatus == noErr && certificateCount > 1)
```

Comment:
I can accept either way here, but why did you choose to do it on a recovery path instead of just putting all the certs (or perhaps cert index 0 and last) into the anchor set always?

Reply:
So far it is only apple.com and icloud.com sites needing this. So I thought I would do it as recovery instead of doing it for all sites - where using just one anchor is good enough. 
My initial test was with putting all certs into anchors and it did work as well. I can update it you think it is better.